pkg/liqo-controller-manager/resource-request-controller/resource-monitors/external-monitor_test.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resourcemonitors
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"google.golang.org/grpc"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+)
+
+type FakeGRPCServer struct {
+	Server *grpc.Server
+	ResourceReaderServer
+}
+
+var grpcSemaphore sync.WaitGroup
+
+func (b *FakeGRPCServer) Start(ctx context.Context) error {
+	address := fmt.Sprintf("%s%d", "127.0.0.1:", 7000)
+	lis, err := net.Listen("tcp", address)
+	Expect(err).ToNot(HaveOccurred())
```

Comment:
You should return the error here, and then assert it in the test statement. Look also at the other functions